### PR TITLE
Active record validations

### PR DESCRIPTION
**Pull request by Krutik31**

### CODE REVIEW

File: app/models/faculty.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+
+  private
+
+  def validate_date_of_birth
+    errors.add(:dob, "birthdate can't be in future.")
+  end
+
+  def check_date?
+    return if (self.dob.blank?)
```
Comment (purvak-simform) [inline app/models/faculty.rb]: paraenthesis is not required for if block

File: app/models/student.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  private
+
+  def validate_date_of_birth
+    errors.add(:dob, "birthdate can't be in future.")
+  end
+
+  def check_date?
+    return if (self.dob.blank?)
+    (Date.parse(self.dob.to_s) > Date.today)
```
Comment (purvak-simform) [inline app/models/student.rb]: same remove parenthesis

File: app/views/faculties/edit.html.erb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  <h2 class="text-center">Update Informations</h2>
+  <h3 class="text-center">(<%= @faculty.first_name %> <%= @faculty.last_name %>)</h3>
+
+  <div class="row">
+    <div class="col-3">
+    </div>
+    <div class="col-6">
+
+      <% if @faculty.errors.any? %>
```
Comment (purvak-simform) [inline app/views/faculties/edit.html.erb]: Align properly

File: app/views/faculties/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+                <%= link_to "Edit", edit_faculty_path(faculty), class:"btn btn-warning btn-sm" %>
+                &nbsp;&nbsp;
+                <%= link_to "Destroy", faculty_path(faculty), data: {turbo_method: :delete, turbo_confirm: "Are you sure?"}, class:"btn btn-danger btn-sm" %>
+              </td>
+            </tr>
+          <% end %>
+        </tbody>
+      </table>
+      <div class="row col-12 justify-content-center"> 
+        <%= link_to "Add new Faculty", new_faculty_path(), class:"btn btn-primary float-left" %>
```
Comment (purvak-simform) [inline app/views/faculties/index.html.erb]: there is no need for empty parenthesis in url

File: app/views/students/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  </div>
+
+  <div class="form-group">
+    <%= form.label :dob, "Date of Birth" %>
+    <%= form.date_field :dob, class:"form-control" %>
+  </div>
+
+  <div class="form-group">
+    <%= form.label :department %>
+    <%= form.select :department, Student::DEPTS, {}, {class: "form-control"}%>
```
Comment (purvak-simform) [inline app/views/students/_form.html.erb:19]: Student::DEPTS What this means here?

**Review by purvak-simform — CHANGES_REQUESTED** [review]

File: app/views/students/_form.html.erb (hunk 5 above)
Comment (Krutik31) [reply app/views/students/_form.html.erb:19]: This will get the DEPTS enum initialized in Student model.

File: app/models/student.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  validates :terms_of_usage, acceptance: { message: "You cannot proceed without accepting Terms of Usage" }
+
+  private
+
+  def validate_date_of_birth
+    errors.add(:dob, "birthdate can't be in future.")
+  end
+
+  def check_date?
+    return if self.dob.blank?
```
Comment (purvak-simform) [inline app/models/student.rb]: check the requirement of self here, update it

**Review by purvak-simform — COMMENTED** [review]
Make commented changes other then that eveything LGTM

**Pull request merged**